modules/service/src/main/scala/lucuma/odb/graphql/mapping/VisitMapping.scala

```diff
@@ -129,41 +116,4 @@ trait VisitMapping[F[_]] extends VisitTable[F]
       selectWithOffsetAndLimit(rOFFSET, rLIMIT, ExecutionEventType, "id", Predicates.executionEvent.id, Predicates.executionEvent.observation.program)
   }
 
-  // WIP.  At the moment the `invoiceHandler` calls the time accounting service
-  // on demand and returns an "invoice" with just the whole execution time.
-  private lazy val invoiceHandler: EffectHandler[F] =
-    new EffectHandler[F] {
-
-      def calculateInvoice(vid: Visit.Id): F[Result[Json]] =
-        services.useTransactionally {
-          timeAccountingService.initialState(vid).map { tas =>
-            val ct = tas.charge.asJson
-            Json.obj(
-              "executionTime" -> ct,
-              "discounts"     -> Json.arr(),
-              "corrections"   -> Json.arr(),
-              "finalCharge"   -> ct
-            ).success
-          }
-        }
-
-      override def runEffects(queries: List[(Query, Cursor)]): F[Result[List[Cursor]]] =
-        (for {
-          vids <- ResultT(queries.traverse { case (_, cursor) => cursor.fieldAs[Visit.Id]("id") }.pure[F])
-          invs <- vids.distinct.traverse { case vid =>
-                    ResultT(calculateInvoice(vid)).tupleLeft(vid)
-                  }
-          res <- ResultT(
-                   vids
-                     .flatMap { vid => invs.find(r => r._1 === vid).map(_._2).toList }
-                     .zip(queries)
-                     .traverse { case (result, (query, parentCursor)) =>
-                       Query.childContext(parentCursor.context, query).map { childContext =>
-                         CirceCursor(childContext, result, Some(parentCursor), parentCursor.fullEnv)
-                       }
-                     }.pure[F]
```

Comment:
With the time accounting data that is currently ready, this `CirceCursor` / `Json` result approach would work fine. Once the discounts and corrections are added though, there will be fields that require database lookups depending on the query.